web/src/reactions.ts

```diff
@@ -523,31 +632,47 @@ export function update_user_fields(clean_reaction_object, should_display_reactor
 
     // The vote_text field set here is used directly in the Handlebars
     // template for rendering (or rerendering!) a message.
-    clean_reaction_object.vote_text = get_vote_text(clean_reaction_object, should_display_reactors);
+    clean_reaction_object.vote_text = get_vote_text(
+        clean_reaction_object.user_ids,
+        should_display_reactors,
+    );
 }
 
-export function get_vote_text(clean_reaction_object, should_display_reactors) {
+export function get_vote_text(user_ids: number[], should_display_reactors: boolean): string {
     if (should_display_reactors) {
-        return comma_separated_usernames(clean_reaction_object.user_ids);
+        return comma_separated_usernames(user_ids);
     }
-    return `${clean_reaction_object.user_ids.length}`;
+    return `${user_ids.length}`;
+}
+
+type ReactionUserIdAndCount = {
+    count?: number;
+    user_ids: number[];
+};
+
+function get_reaction_counts_and_user_ids(message: Message): ReactionUserIdAndCount[] {
+    return [...message.clean_reactions.values()].map((reaction) => ({
+        count: reaction.count,
+        user_ids: reaction.user_ids,
+    }));
 }
 
-function check_should_display_reactors(cleaned_reactions) {
+function check_should_display_reactors(reaction_data: ReactionUserIdAndCount[]): boolean {
```

Comment:
this is part of the second change, so that we can calculate this before finishing building the cleaned reactions